src/main/java/edu/iu/terracotta/service/app/impl/TreatmentServiceImpl.java

```diff
@@ -147,22 +158,27 @@ public TreatmentDto toDto(Treatment treatment, boolean submissions, boolean addA
     }
 
     @Override
-    public Treatment fromDto(TreatmentDto treatmentDto) throws DataServiceException{
-        Treatment treatment = new Treatment();
+    public Treatment fromDto(TreatmentDto treatmentDto, Treatment treatment) throws DataServiceException{
+        if (treatment == null) {
+            treatment = new Treatment();
+        }
+
         treatment.setTreatmentId(treatmentDto.getTreatmentId());
         Optional<Assignment> assignment = allRepositories.assignmentRepository.findById(treatmentDto.getAssignmentId());
 
         if (!assignment.isPresent()) {
             throw new DataServiceException(TextConstants.NO_ASSIGNMENT_IN_TREATMENTDTO);
         }
 
+        entityManager.detach(assignment.get());
```

Comment:
Here and on line 181, why are these entities detached from the persistence context?

For that matter, I don't see why the treatment is detached on line 126 either.

I think it would be worth a comment in these three places why the detach is necessary.

Reply:
I was receiving "this entity was already updated in another transaction" errors on the save() when saving the treatment with the assignment and condition added... It was my (rather weak, I admit) attempt at fixing this quickly.

I was short of time this morning. I will have more time to look deeper into the JPA errors related to that tomorrow morning, if you prefer.

I agree, it's not elegant... :)